src/main/java/redis/clients/jedis/DefaultJedisSocketFactory.java

```diff
@@ -105,6 +111,11 @@ public Socket createSocket() throws JedisConnectionException {
     }
   }
 
+  @Override
+  public void updateHostAndPort(HostAndPort hostAndPort) {
```

Comment:
Should we use `synchronized` here?

Reply:
why? if you're using `volatile` you don't need.

Reply:
@gkorland If we had check-and-set, we should have. But we don't have it here. So it's okay.
(sorry, for a second I forgot that CAS is absent)